virttest/utils_libvirtd.py

```diff
@@ -29,24 +30,51 @@ class Libvirtd(object):
     Class to manage libvirtd service on host or guest.
     """
 
-    def __init__(self, session=None):
+    def __init__(self, service_name=None, session=None):
         """
         Initialize an service object for libvirtd.
 
+        :params service_name: Service name such as virtqemud or libvirtd.
         :params session: An session to guest or remote host.
         """
         self.session = session
-
         if self.session:
             self.remote_runner = remote.RemoteRunner(session=self.session)
             runner = self.remote_runner.run
         else:
             runner = process.run
 
+        self.daemons = []
+        self.service_list = []
+
         if LIBVIRTD is None:
             logging.warning("Libvirtd service is not available in host, "
                             "utils_libvirtd module will not function normally")
-        self.libvirtd = service.Factory.create_service(LIBVIRTD, run=runner)
+
+        self.service_name = service_name
+        if self.service_name is None:
+            self.service_name = "libvirtd"
+        if libvirt_version.version_compare(5, 6, 0, self.session):
+            if libvirtd_decorator.get_libvirtd_split_enable_bit():
+                if self.service_name in ["libvirtd", "libvirtd.service"]:
+                    self.service_list = ['virtqemud', 'virtproxyd',
+                                         'virtnetworkd', 'virtinterfaced',
+                                         'virtnodedevd', 'virtsecretd',
+                                         'virtstoraged', 'virtnwfilterd']
+                elif self.service_name == "libvirtd.socket":
+                    self.service_name = "virtqemud.socket"
+                elif self.service_name in ["libvirtd-tcp.socket", "libvirtd-tls.socket"]:
+                    self.service_name = re.sub("libvirtd", "virtproxyd",
```

Comment:
Do you only want 'virtproxyd.socket'? If that, is it simple to use 'self.service_name = ''virtproxyd.socket'?

Reply:
No, it needs virtproxyd.socket, virtproxyd-tls.socket and virtproxyd-tcp.socket.

Reply:
OK